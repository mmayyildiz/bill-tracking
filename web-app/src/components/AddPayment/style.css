.form {
   width: 80%;
   margin: auto;
   margin-top: 20px;
}

.form-text-input{
   border: none;
   border-bottom: 1px solid #102A58;
   font-size: 18px;
   height: 50px;
   width: 100%;
   margin-top:15px;
}

.btn-container{
  margin-top:50px;  
  text-align: center;
}

.btn{
 padding: 1rem 10rem;
 font-size: 1.2rem;
 border: none;
 border-radius: 10rem;
 cursor: pointer;
}

.btn-green{
  background: #00cd50;
  color: #fff;
}