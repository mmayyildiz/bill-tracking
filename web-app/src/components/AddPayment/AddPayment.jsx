import React, {useState} from 'react';
import { connect } from 'react-redux';
import {createPayment} from '../../store/paymentActions'
import Header from '../Header/Header';
import Message from '../Message/Message';
import './style.css'

const AddPayment = ({ createPayment, history }) => {

    const [newPayment, setNewPayment] = useState({name: '', amount:'', frequency: 'Monthly', startDate: ''});

    const headerProps = {
        title: "Add A Bill",
        subTitle: "Enter your details",
        description: "Keep track of your household spending by adding your bills"
    }

    const onChange = (e) => {
   
        if(e.target.name === 'amount'){

          if(e.target.value === '£'){
            const obj = {[e.target.name]: ''};
            setNewPayment((prevPayment) => ({ ...prevPayment, ...obj }));
          } else{
            const inputVal = (e.target.value.length > 1 && (e.target.value).substring(1)) || e.target.value;
            const isNumber = e.target.value !== '' && !isNaN(inputVal);
             if(isNumber) {
                 const obj = {[e.target.name]: '£'+ inputVal};
                 setNewPayment((prevPayment) => ({ ...prevPayment, ...obj }));
             }
          }

        } else{
          const value = {[e.target.name]: e.target.value};
          setNewPayment((prevPayment) => ({ ...prevPayment, ...value }));
        }
     
    }

    const onSubmit = (e) => {

        e.preventDefault();
        const isValid = newPayment.name && newPayment.amount && newPayment.startDate && newPayment.frequency;
          
        if(isValid){
          const payment = {
            name: newPayment.name,
            amount: newPayment.amount,
            startDate: newPayment.startDate, 
            frequency: newPayment.frequency
          };
          createPayment(payment, history);
        }else{
          const message = {message: 'All fields must be filled!'}
          setNewPayment((prevPayment) => ({ ...prevPayment, ...message }));
        }
    }

    const {name, amount, startDate, frequency, message} = newPayment;

    return (
        <React.Fragment>
                <Header {...headerProps} />
                <div className="content">
                  <div className="form">
                    <input className="form-text-input" type="text" name="name" placeholder="Name" onChange={onChange} value={name}/>
                    <input className="form-text-input" type="text" name="amount" placeholder="Amount" onChange={onChange} value={amount} />
                    <input className="form-text-input" type="date" name="startDate" placeholder="Start Date" onChange={onChange} value={startDate}/>
                    <select className="form-text-input" name="frequency" value={frequency} placeholder="Frequency" onChange={onChange}>
                      <option value="Weekly">Weekly</option>
                      <option value="Monthly">Monthly</option>
                      <option value="Annually">Annually</option>
                    </select>
                    <div className="btn-container">
                        <input type="button" className="btn btn-green" onClick={onSubmit} value="Add new payment" />
                    </div>
                  </div>
                  { message && <Message msg={message} /> }
                </div>
        </React.Fragment>
    );
}

const mapStateToProps = state => ({
    errors: state.errors
  });
  

export default connect(mapStateToProps, { createPayment })(AddPayment);
