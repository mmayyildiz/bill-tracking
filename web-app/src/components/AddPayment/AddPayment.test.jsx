import React from 'react';
import AddPayment from './AddPayment';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import renderer from 'react-test-renderer';


const mockStore = configureStore();

const mockCallFunction = jest.fn();

const store = mockStore({
  createPayment: mockCallFunction
});

const wrapper = renderer.create(
  <Provider store={store}>
    <AddPayment />
  </Provider>
);

describe('AddPayment', () => {

   it('should render with given state from Redux store', () => {
    expect(wrapper.toJSON()).toMatchSnapshot();
   });

});
