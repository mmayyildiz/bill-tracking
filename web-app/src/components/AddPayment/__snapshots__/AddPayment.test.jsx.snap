// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddPayment should render with given state from Redux store 1`] = `
Array [
  <div
    className="header"
  >
    <p>
        
      Add A Bill
       
    </p>
  </div>,
  <div
    className="sub-header"
  >
    <p
      className="sub-header-title"
    >
       
      Enter your details
       
    </p>
    <p
      className="sub-header-title-description"
    >
       
      Keep track of your household spending by adding your bills
    </p>
  </div>,
  <div
    className="content"
  >
    <div
      className="form"
    >
      <input
        className="form-text-input"
        name="name"
        onChange={[Function]}
        placeholder="Name"
        type="text"
        value=""
      />
      <input
        className="form-text-input"
        name="amount"
        onChange={[Function]}
        placeholder="Amount"
        type="text"
        value=""
      />
      <input
        className="form-text-input"
        name="startDate"
        onChange={[Function]}
        placeholder="Start Date"
        type="date"
        value=""
      />
      <select
        className="form-text-input"
        name="frequency"
        onChange={[Function]}
        placeholder="Frequency"
        value="Monthly"
      >
        <option
          value="Weekly"
        >
          Weekly
        </option>
        <option
          value="Monthly"
        >
          Monthly
        </option>
        <option
          value="Annually"
        >
          Annually
        </option>
      </select>
      <div
        className="btn-container"
      >
        <input
          className="btn btn-green"
          onClick={[Function]}
          type="button"
          value="Add new payment"
        />
      </div>
    </div>
  </div>,
]
`;
